### PR TITLE
Add comprehensive unit and integration tests for CurrencyService and UserCurrencyService

### PR DESCRIPTION
**Pull request by qhasib-tootisabz**
- Implemented unit tests for all methods in CurrencyService and UserCurrencyService.
- Developed end-to-end integration tests covering Controller, Service, and Repository layers.
- Added edge case tests to ensure robustness and proper handling of exceptional scenarios.
- Verified that all tests passed successfully, ensuring the reliability and correctness of the services.

### CODE REVIEW

File: services-users/src/main/java/tootisabz/servicesusers/Application/Service/UserCurrencyService.java
Comment (Zabihullah-Ibrahimkhail) [inline services-users/src/main/java/tootisabz/servicesusers/Application/Service/UserCurrencyService.java:1]: Make you code clean and remove un necessary comments

File: services-users/src/main/java/tootisabz/servicesusers/UI/Controller/UserCurrencyController.java [hunk 1]
```diff
@@ -23,27 +22,59 @@ public class UserCurrencyController {
...
         List<UserCurrency> userCurrencies = userCurrencyService.getUserCurrencies(userId);
-        return new ResponseEntity<>(userCurrencies, HttpStatus.OK);
+        return new ResponseEntity<>(userCurrencies, HttpStatus.OK); // Always return the list, even if empty
     }
 
+//    public ResponseEntity<List<UserCurrency>> getUserCurrencies(@PathVariable UUID userId) {
+//        List<UserCurrency> userCurrencies = userCurrencyService.getUserCurrencies(userId);
+//        if (userCurrencies.isEmpty()) {
+//            return new ResponseEntity<>(HttpStatus.OK);
+//        }
```
Comment (Zabihullah-Ibrahimkhail) [inline services-users/src/main/java/tootisabz/servicesusers/UI/Controller/UserCurrencyController.java:40]: remove the comments

File: services-users/src/test/java/tootisabz/servicesusers/ServicesUsersApplicationTests.java [hunk 2]
```diff
@@ -8,6 +8,7 @@ class ServicesUsersApplicationTests {
 
 	@Test
 	void contextLoads() {
+
```
Comment (Zabihullah-Ibrahimkhail) [inline services-users/src/test/java/tootisabz/servicesusers/ServicesUsersApplicationTests.java:11]: need clarification 
complete the code or if you don't need it remove the method

File: services-users/src/test/java/tootisabz/servicesusers/UI/Controller/UserCurrencyControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                        .contentType("application/json")
+                        .content(objectMapper.writeValueAsString(invalidCurrencyDTO)))
+                .andExpect(status().isBadRequest())  // Expect 400 Bad Request
+                .andExpect(jsonPath("$.error").value("UserId or CurrencyId cannot be null"));  // Validate error message
+    }
+
+    // Unauthorized Access (missing token)
+//    @Test
+//    void testUnauthorizedAccess() throws Exception {
+//        // Act & Assert
```
Comment (Zabihullah-Ibrahimkhail) [inline services-users/src/test/java/tootisabz/servicesusers/UI/Controller/UserCurrencyControllerTest.java:180]: remove not needed comments

File: src/test/java/com/tootisabz/CrystalPay/CrystalPayApplicationTests.java [hunk 4]
```diff
@@ -10,4 +10,10 @@ class CrystalPayApplicationTests {
 	void contextLoads() {
 	}
 
+	@Test
+	void Currencies(){
+
```
Comment (Zabihullah-Ibrahimkhail) [inline src/test/java/com/tootisabz/CrystalPay/CrystalPayApplicationTests.java:15]: complete it or remove it

**Review by Zabihullah-Ibrahimkhail — CHANGES_REQUESTED** [review]
see the comments and do tasks accordingly

File: src/test/java/com/tootisabz/CrystalPay/CrystalPayApplicationTests.java (hunk 4 above)
Comment (qhasib-tootisabz) [reply src/test/java/com/tootisabz/CrystalPay/CrystalPayApplicationTests.java:15]: Removed it successfully.

File: services-users/src/test/java/tootisabz/servicesusers/UI/Controller/UserCurrencyControllerTest.java (hunk 3 above)
Comment (qhasib-tootisabz) [reply services-users/src/test/java/tootisabz/servicesusers/UI/Controller/UserCurrencyControllerTest.java:180]: I had commented it out earlier while testing other methods, but I’ve now uncommented it to ensure the test properly checks for unauthorized access to the currencies endpoint

File: services-users/src/test/java/tootisabz/servicesusers/ServicesUsersApplicationTests.java (hunk 2 above)
Comment (qhasib-tootisabz) [reply services-users/src/test/java/tootisabz/servicesusers/ServicesUsersApplicationTests.java:11]: This is a default test generated by Spring Boot when creating a new application.
It ensures that the Spring application context loads without any issues.
It's a simple way to verify that the basic configuration of the application is correct and all required beans are properly set up.

File: services-users/src/main/java/tootisabz/servicesusers/UI/Controller/UserCurrencyController.java (hunk 1 above)
Comment (qhasib-tootisabz) [reply services-users/src/main/java/tootisabz/servicesusers/UI/Controller/UserCurrencyController.java:40]: Removed unnecessary comment successfully.

File: services-users/src/main/java/tootisabz/servicesusers/Application/Service/UserCurrencyService.java
Comment (qhasib-tootisabz) [reply services-users/src/main/java/tootisabz/servicesusers/Application/Service/UserCurrencyService.java:1]: Cleaned the code and removed unnecessary comments.
